fault/tester.py

```diff
@@ -53,3 +57,31 @@ def serialize(self):
     def compile_and_run(self, target="verilator", **kwargs):
         target_inst = self.make_target(target, **kwargs)
         target_inst.run()
+
+    def __check_interfaces_match(self, old_circuit, new_circuit):
+        old_port_names = old_circuit.interface.ports.keys()
+        for name in old_port_names:
+            if not hasattr(new_circuit, name):
```

Comment:
More of a style thing, but I've being moving away from using `hasattr` in favor of `name in new_circuit.interface.ports`. I think it is cleaner and more robust. Same throughout this change.